src/main/java/Duke.java

```diff
@@ -1,10 +1,346 @@
+    }
+
+
+
+    public static void createEvent() throws DukeException{
+        if(!userInput.contains("/at") || userInput.length()<= 5){
+            throw new DukeException("☹ OOPS!!! The description or the date of an event cannot be empty.");
+        }
+        description = userInput.substring(6);
+        int currentTaskIndex = listLength;
+        int slashIndex = userInput.indexOf("/");
+        int slashAtIndex = userInput.indexOf("/at");
+        command = userInput.substring(6,slashIndex);
+        String at = userInput.substring(slashAtIndex+4);
+        Event event = new Event(command,at);
+        tasks.add(event);
+        //list[currentTaskIndex] = new Event(command,at);
+        listLength++;
+        System.out.println(SEPARATOR);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(event);
+        System.out.println("Now you have " + listLength + " tasks in the list");
+        System.out.println(SEPARATOR);
+        saveFile();
+    }
+
+        public static void saveFile(){
+            File saveDirection = new File("data");
+            saveDirection.mkdir();
+            File saveFile = new File(saveDirection,"duke.txt");
+            try{
+                saveFile.createNewFile();
+                FileWriter fileWriter = new FileWriter(saveFile);
+                for(int i = 0; i < listLength;i++){
+                    char taskType = tasks.get(i).toString().charAt(1);
+                    fileWriter.write(taskType + "||" + tasks.get(i).isDone() + "||" + tasks.get(i).getTaskCommand());
+                    if (taskType == 'D'){
+                        fileWriter.write("||" + ((Deadline) tasks.get(i)).getBy());
+                    } else if (taskType == 'E'){
+                        fileWriter.write("||" + ((Event) tasks.get(i)).getEventTime());
+                    }
+                    fileWriter.write("\n");
+                }
+                fileWriter.close();
+            } catch (IOException e){
+                e.printStackTrace();
+            }
+
+        }
+
+
+        public static void setTasks(){
+             tasks = new ArrayList<Task>();
+             listLength = 0;
+             openTaskFromFile();
+        }
+
+
+        public static Task createSavedTask(String line){
+            String[] lineDivision = line.split("\\|\\|");
+            Task saveTask;
+            if(lineDivision[0].equals("D")){
+
+                saveTask = new Deadline(lineDivision[2],lineDivision[3],Boolean.parseBoolean(lineDivision[1]));
+
+            } else if (lineDivision[0].equals("E")){
+
+                saveTask = new Event(lineDivision[2],lineDivision[3],Boolean.parseBoolean(lineDivision[1]));
+
+            } else {
+
+                saveTask = new Todo(lineDivision[2],Boolean.parseBoolean(lineDivision[1]));
+
+            }
+            return saveTask;
+        }
+
+
+        public static void openTaskFromFile(){
+            File saveDirection = new File("data");
```

Comment:
`saveDirection`  is an action. Perhaps [rename this to a noun](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-nouns-for-things-and-verbs-for-actions) for your file object?